### PR TITLE
Phase2-hgx102 Make second step  to define post TDR geometry

### PR DESCRIPTION
**Pull request by bsunanda**
Corrected bug for the EE part and added code + xml for the HE part (full silicon and mixed layers)

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/plugins/DDHGCalHEAlgo.cc [hunk 1]
```diff
@@ -79,163 +169,380 @@ void DDHGCalHEAlgo::initialize(const DDNumericArguments & nArgs,
...
+				    << layerThick_[i];
+#endif
+      DDName matName(DDSplit(materials_[ii]).first, 
+		     DDSplit(materials_[ii]).second);
+      DDMaterial matter(matName);
+      DDLogicalPart glog;
+      if (layerSense_[ly] == 0) {
+	double alpha = CLHEP::pi/sectors_;
+	double rmax  = routF*cos(alpha) - tol;
+	std::vector<double> pgonZ, pgonRin, pgonRout;
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/plugins/DDHGCalHEAlgo.cc]: do these need to be vectors? if they only have 2 elements each, they could just be `std::array`s.

**Comment by davidlange6** [comment]
please test

**Comment by kpedro88** [comment]
+1

File: Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc [hunk 2]
```diff
@@ -131,7 +130,7 @@ void DDHGCalEEAlgo::execute(DDCompactView& cpv) {
   copies_.clear();
```
Comment (ianna) [inline Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc]: copies_ is used only for debugging. Please, move it.

File: Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc [hunk 3]
```diff
@@ -270,10 +276,11 @@ void DDHGCalEEAlgo::positionSensitive(DDLogicalPart& glog, double rin,
   double dy   = 0.75*R;
   int    N    = (int)(0.5*rout/r) + 2;
   int    ium(0), ivm(0), iumAll(0), ivmAll(0), kount(0), ntot(0), nin(0);
-  std::vector<int>  ntype(3,0);
+  std::vector<int>  ntype(6,0);
```
Comment (ianna) [inline Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc]: ntype is used only in debug mode. Please, move it to in between the preprocessor directives.

**Review by ianna — CHANGES_REQUESTED** [review]

File: Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc [hunk 4]
```diff
@@ -160,6 +159,7 @@ void DDHGCalEEAlgo::constructLayers(const DDLogicalPart& module,
     int     laymax = laymin+layers_[i];
     double  zz     = zi;
     double  thickTot(0);
+    std::vector<double> pgonZ(2), pgonRin(2), pgonRout(2);
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc]: I think the call to polyhedra wants std::vector

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc (hunk 2 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc]: Taken care of this

**Comment by bsunanda** [comment]
@ianna @kpedro88 Please approve this

**Comment by kpedro88** [comment]
+1

**Comment by ianna** [comment]
+1

**Comment by fabiocos** [comment]
+1

pure upgrade code

**Pull request merged**